src/Sylius/Bundle/CoreBundle/Processor/CatalogPromotionStateProcessor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Processor;
+
+use SM\Factory\FactoryInterface;
+use Sylius\Bundle\CoreBundle\Checker\CatalogPromotionEligibilityCheckerInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionTransitions;
+
+final class CatalogPromotionStateProcessor implements CatalogPromotionStateProcessorInterface
+{
+    public function __construct(
+        private CatalogPromotionEligibilityCheckerInterface $catalogPromotionEligibilityChecker,
+        private FactoryInterface $stateMachine
+    ) {
+    }
+
+    public function process(CatalogPromotionInterface $catalogPromotion): void
+    {
+        $stateMachine = $this->stateMachine->get($catalogPromotion, CatalogPromotionTransitions::GRAPH);
+
+        if ($stateMachine->can(CatalogPromotionTransitions::TRANSITION_PROCESS)) {
+            $stateMachine->apply(CatalogPromotionTransitions::TRANSITION_PROCESS);
```

Comment:
As I spoke with @CoderMaggie about it and for now we should have only 2 states of Catalog Promotion: `active` and `inactive`, more states probably we will add in the future.
with state `process` (same for `failed`) is a problem because our current implementation applies for catalog promotion on Variant per variant instead of per Catalog Promotion (it was an old approach). more states can be introduced but after more advanced research.

Reply:
Ok, fixed.